community/cypher/frontend-3.3/src/main/scala/org/neo4j/cypher/internal/frontend/v3_3/parser/Clauses.scala

```diff
@@ -66,21 +66,29 @@ trait Clauses extends Parser
     CreatedGraphAt | CreatedGraphAs
 
   private def CreatedGraphAt: Rule1[(Boolean,  ast.Variable, Option[ast.Pattern], ast.GraphUrl)] =
-    Snapshot ~~ keyword("GRAPH") ~~ Variable ~~ optional(keyword("OF") ~~ Pattern) ~~ keyword("AT") ~~ GraphUrl ~~> {
+    OptSnapshot ~~ keyword("GRAPH") ~~ Variable ~~ optional(keyword("OF") ~~ Pattern) ~~ keyword("AT") ~~ GraphUrl ~~> {
       (snapshot, graph, pattern, url) => (snapshot, graph, pattern, url)
     }
 
   private def CreatedGraphAs: Rule1[(Boolean,  ast.Variable, Option[ast.Pattern], ast.GraphUrl)] =
-    Snapshot ~~ keyword("GRAPH") ~~ optional(keyword("OF") ~~ Pattern) ~~ keyword("AT") ~~ GraphUrl ~~ keyword("AS") ~~ Variable ~~> {
+    OptSnapshot ~~ keyword("GRAPH") ~~ optional(keyword("OF") ~~ Pattern) ~~ keyword("AT") ~~ GraphUrl ~~ keyword("AS") ~~ Variable ~~> {
       (snapshot, pattern, url, graph) => (snapshot, graph, pattern, url)
     }
 
+  private def OptSnapshot: Rule1[Boolean] = rule("[SNAPSHOT]") {
+    optional(keyword("SNAPSHOT") ~~ push(true)) ~~>(_.getOrElse(false))
+  }
+
   def Persist: Rule1[ast.Persist] = rule("PERSIST") {
-    keyword("PERSIST") ~~ Snapshot ~~ BoundGraph ~~ keyword("TO") ~~ GraphUrl ~~>>(ast.Persist(_, _, _))
+    keyword("PERSIST") ~~ BoundGraph ~~ keyword("TO") ~~ GraphUrl ~~>>(ast.Persist(_, _))
+  }
+
+  def Snapshot: Rule1[ast.Snapshot] = rule("SNAPSHOT") {
```

Comment:
So this is a change in syntax, then. We're now looking at 

```
SNAPSHOT foo TO $url // persists a snapshot of foo at $url
PERSIST foo TO $url // persists foo as-is to $url
RELOCATE foo TO $url // unbinds foo from it's previous address (if any) and binds it to $url
```

And snapshots are using copy-semantics rather than share-semantics.